tests/util/test_config.py

```diff
@@ -0,0 +1,151 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any, Dict, Set
+
+from chia.server.outbound_message import NodeType
+from chia.types.peer_info import UnresolvedPeerInfo
+from chia.util.config import get_unresolved_peer_infos
+from chia.util.ints import uint16
+from tests.util.misc import DataCase, Marks, datacases
+
+
+@dataclass
+class GetUnresolvedPeerInfosCase(DataCase):
+    description: str
+    service_config: Dict[str, Any]
+    requested_node_type: NodeType
+    expected_peer_infos: Set[UnresolvedPeerInfo]
+    marks: Marks = ()
+
+    @property
+    def id(self) -> str:
+        return self.description
```

Comment:
The `@property` was just to make a calculated description in the case you probably looked at.  Since we're just using the string directly it can be an `id` attribute directly instead of `description` plus the `id` property.

Reply:
I tried that first, but i got the following warning:
![grafik](https://github.com/Chia-Network/chia-blockchain/assets/5189997/65949efe-7225-4d1d-b3f4-be5cfb723811)

I also noticed other cases do not inherit from `DataCase`, in which case the id prop wont result in a warning, but the whole `@datacases()` section gets flagged instead:
![grafik](https://github.com/Chia-Network/chia-blockchain/assets/5189997/2022e7fd-455c-424a-874e-52ff01f36e94)

Reply:
Yeah, if you want to keep it this way we can.  There are various other cases where I did do `id: str` and I just deal with the pycharm complaint.  I like pycharm 'working' right but I mypy is the actual blocker in CI.  Feel free to either change it or leave it and just resolve this conversation as you see fit.

Reply:
Is there a specific reason why `DataCase` does not look like this?
![grafik](https://github.com/Chia-Network/chia-blockchain/assets/5189997/94a80775-b326-474a-b147-f9123e589a5c)

Reply:
This would allow this (`id` override because its easier than a custom init with super call):
![grafik](https://github.com/Chia-Network/chia-blockchain/assets/5189997/fcdebce0-41e2-4ae1-8083-741c03332448)

Reply:
Thoughts on this @altendky ?

Reply:
Changed it, if not desired we can remove ofc 👍

Reply:
Actually reverted it again (so its now as it was before), because with the changes to DataCase other classes with a computed id prop failed mypy

It probably makes sense to cleanup and unify DataCase usage and structure in a separate PR

Reply:
Sorry I left this hanging.  Yes, if we're making changes to datacases, let's take that up separately.  Thanks.  I'm trying to remember the details around the property in the protocol.  I totally agree it's funny and pycharm not understanding it is annoying.  But I did try.  `:]`